interpreter.js

```diff
@@ -3556,6 +3556,12 @@ Interpreter.prototype['stepMemberExpression'] = function() {
     if (state.components) {
       stack[stack.length - 1].value = [state.object_, state.value];
     } else {
+      if (state.object_ == this.NULL || state.object_ == this.UNDEFINED)
+      {
+        var name = this.getFullyQualifiedName(node.object);
+        this.throwException(this.TYPE_ERROR, name + " is not defined.");
```

Comment:
I think we should be explicit about how this is thrown for values that are either `null` or `undefined`; "not defined" is a bit vague.

I can see there being value in being explicit about what we're designing these error messages to resemble; link directly to the relevant documentation at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Unexpected_type and make sure that we match them as closely as possible

Reply:
Good catch. It was supposed to be "undefined", but I copied the error message line from elsewhere in the interpreter code and missed updating it. Good call on splitting out undefined/null too. I'll split up the logic to be more clear. From the user side, this won't have much affect, because we throw a different error for null references, but it will be useful for cases that we miss and such.

Note
Error for null references: Oops, we can’t figure out what _myVar_ is - perhaps you meant the string _“myVar”_ with quotes? If this is meant to be a variable, make sure you declared a variable: var _myVar_